python/lsst/ts/standardscripts/auxtel/latiss.py

```diff
@@ -0,0 +1,244 @@
+
+    Parameters
+    ----------
+    atcam : `salobj.Remote(SALPY_ATCam)`
+        Remote for the ATCamera.
+    atspec : `salobj.Remote(SALPY_ATSpectrograph)`
+        Remote for the ATSpectrograph.
+    """
+
+    def __init__(self, atcam, atspec):
+
+        self.atcam = atcam
+        self.atspec = atspec
+
+        self.read_out_time = 2.  # readout time (sec)
+        self.shutter_time = 1  # time to open or close shutter (sec)
+
+        self.cmd_timeout = 30.
+        self.end_readout_timeout = 60.
+
+        self.cmd_lock = asyncio.Lock()
+
+    async def take_bias(self, nbias, checkpoint=None):
+        """Take a series of bias images.
+
+        Parameters
+        ----------
+        nbias : `int`
+            Number of bias frames to take.
+        checkpoint : `coro`
+            A optional coroutine that accepts one string argument that is
+            called before each bias is taken.
+
+        """
+        for i in range(nbias):
+            tag = f'bias_{i+1:04}_{nbias:04}'
+
+            if checkpoint is not None:
+                await checkpoint(tag)
+            await self.expose(exp_time=0., shutter=False,
+                              image_seq_name=tag,
+                              science=True, guide=False, wfs=False)
+
+    async def take_darks(self, exptime, ndarks, checkpoint=None):
+        """Take a series of bias images.
+
+        Parameters
+        ----------
+        exptime : `float`
+            Exposure time for darks.
+        ndarks : `int`
+            Number of dark frames to take.
+        checkpoint : `coro`
+            A optional coroutine that accepts one string argument that is
+            called before each dark is taken.
+
+        """
+        for i in range(ndarks):
+            tag = f'dark_{i+1:04}_{ndarks:04}'
+            if checkpoint is not None:
+                await checkpoint(tag)
+            await self.expose(exp_time=exptime, shutter=False,
+                              image_seq_name=tag,
+                              science=True, guide=False, wfs=False)
+
+    async def take_flats(self, exptime, nflats,
+                         latiss_filter=None, latiss_grating=None, latiss_linear_stage=None,
+                         checkpoint=None):
+        """Take a series of flat field images.
+
+        Parameters
+        ----------
+        exptime : `float`
+            Exposure time for flats.
+        nflats : `int`
+            Number of flat frames to take.
+        latiss_filter : `None` or `int` or `str`
+            Filter id or name. If None, ignore setting filter.
+        latiss_grating : `None` or `int` or `str`
+            Grating id or name.  If None, ignore setting grating.
+        latiss_linear_stage : `None` or `float`
+            Linear stage position.  If None, ignore setting linear stage.
+        checkpoint : `coro`
+            A optional coroutine that accepts one string argument that is
+            called before each flat field image is taken.
```

Comment:
A coroutine is what you get when you call `checkpoint(tag)`

I suggest calling it an "awaitable callback", "asynchronous callback" or something like that.